File: src/main/java/thecudster/sre/core/gui/SimpleButton.java
package thecudster.sre.core.gui;

import net.minecraft.client.audio.SoundHandler;
import net.minecraft.client.gui.GuiButton;

public class SimpleButton extends GuiButton {
    public boolean hasPlayedSound = false;
    public String filterName = null;

    public SimpleButton(int buttonId, int x, int y, String buttonText) {
        super(buttonId, x, y, buttonText);
    }

    public SimpleButton(int buttonId, int x, int y, int widthIn, int heightIn, String buttonText) {
        super(buttonId, x, y, widthIn, heightIn, buttonText);
    }

    public SimpleButton(int buttonId, int x, int y, int widthIn, int heightIn, String buttonText, String buttonName) {
        super(buttonId, x, y, widthIn, heightIn, buttonText);
        this.filterName = buttonName;
    }
    @Override
    public void playPressSound(SoundHandler soundHandlerIn) {
        //if (!hasPlayedSound) {
          //  super.playPressSound(soundHandlerIn);
          //  hasPlayedSound = true;
        //}
    }
}

File: src/main/kotlin/thecudster/sre/core/gui/SimpleButton.kt
package thecudster.sre.core.gui

import net.minecraft.client.gui.GuiButton
import net.minecraft.client.audio.SoundHandler

class SimpleButton : GuiButton {
    var hasPlayedSound = false
    var filterName: String? = null

    constructor(buttonId: Int, x: Int, y: Int, buttonText: String?) : super(buttonId, x, y, buttonText) {}
    constructor(buttonId: Int, x: Int, y: Int, widthIn: Int, heightIn: Int, buttonText: String?) : super(
        buttonId,
        x,
        y,
        widthIn,
        heightIn,
        buttonText
    ) {
    }

    constructor(
        buttonId: Int,
        x: Int,
        y: Int,
        widthIn: Int,
        heightIn: Int,
        buttonText: String?,
        buttonName: String?
    ) : super(buttonId, x, y, widthIn, heightIn, buttonText) {
        filterName = buttonName
    }

    override fun playPressSound(soundHandlerIn: SoundHandler) {
        //if (!hasPlayedSound) {
        //  super.playPressSound(soundHandlerIn);
        //  hasPlayedSound = true;
        //}
    }
}